openmc/filter.py

```diff
@@ -300,7 +300,43 @@ def get_bin_index(self, filter_bin):
 
         """
 
-        if filter_bin not in self.bins:
+        try:
+            # Filter bins for a mesh are an (x,y,z) tuple
+            if isinstance(self, MeshFilter):
+                # Convert (x,y,z) to a single bin -- this is similar to
+                # subroutine mesh_indices_to_bin in openmc/src/mesh.F90.
+                if len(self.mesh.dimension) == 3:
+                    nx, ny, nz = self.mesh.dimension
+                    val = (filter_bin[0] - 1) * ny * nz + \
+                          (filter_bin[1] - 1) * nz + \
+                          (filter_bin[2] - 1)
+                else:
+                    nx, ny = self.mesh.dimension
+                    val = (filter_bin[0] - 1) * ny + \
+                          (filter_bin[1] - 1)
+
+                filter_index = val
```

Comment:
I would prefer you get rid of all the `isinstance` blocks and just override `get_bin_index` on the relevant subclasses.